src/library/scala/math/package.scala

```diff
@@ -435,4 +439,21 @@ package object math {
   /** @group exact */
   def toIntExact(x: Long): Int = java.lang.Math.toIntExact(x)
 
+  /** @group clamp */
+  def clamp(x: Byte, lower: Byte, upper: Byte): Byte = max(lower, min(x, upper)).toByte
```

Comment:
there are several possible parameter orders that can be argued, but I opine that the value and the bounds should be in different parameter lists, so that if you get the parameter order wrong, it doesn't compile. one of:

```scala
def clamp(x: Int)(lower: Int, upper: Int): Int
def clamp(lower: Int, upper: Int)(x: Int): Int
```

there is also of course the argument that the value ought to go between the bounds it's being clamped between, though I imagine that would be error-prone:

```scala
def clamp(lower: Int, x: Int, upper: Int): Int
```

Reply:
Within suggested signatures, `def clamp(lower: Int, upper: Int)(x: Int): Int` sounds nice since it allow us to write something like `val range = clamp(0, 100)` without `_`.
Do such multiple parameter lists have overhead compared to single parameter list ?

Reply:
as long as you don't partially apply it, they're indistinguishable at the JVM level